ingest/cli_parser.py

```diff
@@ -74,6 +74,14 @@ def validate_arguments(parsed_args):
             raise ValueError(
                 f" Invalid argument: unable to connect to a BigQuery table called {parsed_args.bq_table}."
             )
+    if (
+        "differential_expression" in parsed_args
```

Comment:
Why do we need this in addition to the `--differential-expression` kwarg?  They seem redundant.

Reply:
If I only check for the annotation_type kwarg, that argument isn't set for other types of ingest pipeline runs and all other non-DE ingest jobs would fail with:
AttributeError: 'Namespace' object has no attribute 'annotation_type'

Checking for "differential_expression" in parsed_args, ensures I'm only doing this validation for DE jobs.

Reply:
Maybe I didn't make myself clear - apologies.  If we look at the example command line:
```
python ingest_pipeline.py --study-id addedfeed000000000000000 --study-file-id dec0dedfeed1111111111111 differential_expression --annotation-name cluster_annot --annotation-type group --annotation-scope cluster --matrix-file-path ../tests/data/differential_expression/de_integration.tsv --matrix-file-type dense --annotation-file ../tests/data/differential_expression/de_integration_annotated_cluster.txt --cluster-file ../tests/data/differential_expression/de_integration_annotated_cluster.txt --cluster-name de_integration --study-accession SCPdev --differential-expression
```
There is the positional argument of `differential_expression` that you call out here, but then also at the end, there is the `--differential-expression` kwarg.  Why do we need both?  Looking in `cli_parser.py`, I see this:
```
# Differential expression subparsers
parser_differential_expression = subparsers.add_parser(
    "differential_expression",
    help="Indicates differential expression analysis processing",
)

parser_differential_expression.add_argument(
    "--differential-expression",
    required=True,
    action="store_true",
    help="Indicates that differential expression analysis should be invoked",
)
```
They look to me to be functionally equivalent.